### ISSUE TITLE
custom colors table

### ISSUE THREAD

**Issue by pedrohgmacedo** [sontungexpt/sttusline#13]
Hi, I'm trying to replace lualine with your plugin. 
Is there a way to replace the colors.lua table for all components? 
The current theme does not handle light themes very well. 
THanks

**Comment by marcelmanz** [comment]
I would like to know too.

I think [here](https://github.com/sontungexpt/sttusline?tab=readme-ov-file#override-default-component) there is some information on how to do it "manually".

**Comment by pedrohgmacedo** [comment]
> I would like to know too.
> 
> I think [here](https://github.com/sontungexpt/sttusline?tab=readme-ov-file#override-default-component) there is some information on how to do it "manually".

Yeah, but then I would be basically be rewriting all the plugin components. 
For instance, here is how the encoding component configures the colors:
https://github.com/sontungexpt/sttusline/blob/133bb40d249e0167c89bb352ff8442b821fb07e9/lua/sttusline/components/encoding.lua#L1-L4

If I understood it correctly you'd have to do that for every component. 
I was thinking of maybe adding an entry point to override the colors component, and make the plugin source from there if it exists, and if not, source from colors.

**Comment by marcelmanz** [comment]
Yes you are right, that would make life much easier 😆

**Comment by marcelmanz** [comment]
more info [here](https://github.com/sontungexpt/sttusline/issues/1)

**Comment by sontungexpt** [comment]
I'm taking a break after a busy period, finally having time to dive into studying Neovim's API highlight. I'll be concentrating on fixing and optimizing it. I suggest using the table_version branch, as I no longer use the main branch.
